python/ql/src/semmle/python/frameworks/Stdlib.qll

```diff
@@ -1542,6 +1542,52 @@ private module Stdlib {
   class Sqlite3 extends PEP249Module {
     Sqlite3() { this = sqlite3() }
   }
+
+  // ---------------------------------------------------------------------------
+  // re
+  // ---------------------------------------------------------------------------
+  private module Re {
+    /** List of re methods. */
+    private class ReMethods extends string {
+      ReMethods() { this in ["match", "fullmatch", "search", "split", "findall", "finditer"] }
+    }
+
+    /** re.ReMethod(pattern, string) */
+    private class DirectRegex extends DataFlow::CallCfgNode, RegexExecution::Range {
+      DataFlow::Node regexNode;
+
+      DirectRegex() {
+        this = API::moduleImport("re").getMember(any(ReMethods m)).getACall() and
+        regexNode = this.getArg(0)
+      }
+
+      override DataFlow::Node getRegexNode() { result = regexNode }
+    }
+
+    /** re.compile(pattern).ReMethod */
+    private class CompiledRegex extends DataFlow::CallCfgNode, RegexExecution::Range {
+      DataFlow::Node regexNode;
+
+      CompiledRegex() {
+        exists(DataFlow::CallCfgNode patternCall, DataFlow::AttrRead reMethod |
+          this.getFunction() = reMethod and
+          patternCall = API::moduleImport("re").getMember("compile").getACall() and
+          patternCall = reMethod.getObject().getALocalSource() and
+          reMethod.getAttributeName() instanceof ReMethods and
+          regexNode = patternCall.getArg(0)
```

Comment:
A little comment about this behaviour:

As it is set now, the actual sink would be `re.compile(this)`, but as far as I've been testing with real projects and `LGTM` queries, it's a bit tiring searching for the `reMethod` call. Shouldn't the `select` statement in the taint config also show the `reMethod` call?

Reply:
That would be great, especially since you have felt personally that it would be a help. We may have to work a bit to get the information across, if we keep to the standard structure with a configuration in its own file. One idea is to define a class to act as sink and give that class a member predicate to reveal the sql execution. In the select, we can cast the sink to our special class and extract the sql execution.